### ISSUE TITLE
decode() and fromPartial() skip falsy map entries

### ISSUE THREAD

**Issue by timostamm** [stephenh/ts-proto#79]
decode() and fromPartial() skip falsy map entries
-------------------------------------------------

Map entries with `0`, `false` or `""` values are skipped by decode() and fromPartial() 

msg.proto:
```proto
message Msg {
  map<string, int32> map_field = 1;
}
```

Reproduction:
```ts
const a: Msg = {
  mapField: {
    "key" : 0
  }
}

const b = Msg.decode(Msg.encode(a).finish());
b.mapField.key; // -> is undefined, but should be 0

const c = Msg.fromPartial({ mapField: { key: 0 } });
c.mapField.key; // -> is undefined, but should be 0
```

The reason lies in the generated decode() function:

```typescript
const entry1 = Msg_MapFieldEntry.decode(reader, reader.uint32());
if (entry1.value) { // this skips 0 too
  message.mapField[entry1.key] = entry1.value;
}
```

I think the condition should be changed to `entry1.value !== undefined`. Or maybe even throw an error (in decode, not in fromPartial). All tests pass with this change.

Let me know what you think, @stephenh. Can send a PR with test coverage.

**Comment by stephenh** [comment]
Ah shoot, yes, @timostamm that's a silly mistake and a PR would be great. Thanks!

**Comment by stephenh** [comment]
I'm slowly catching up on a few ts-proto things and this seemed easy so went ahead and released a fix as v1.21.5. Thanks for the report!

**Comment by timostamm** [comment]
Have been very busy the last days. Thank you for the quick fix!
